### PR TITLE
Introduce `IsRefasterAsVarargs` matcher

### PR DESCRIPTION
**Pull request by mlrprananta**
A follow up for the XXX of `StreamOfArray` encountered in #619

```
Introduce `IsRefasterAsVarargs` matcher for use by Refaster templates (#623)
```

### CODE REVIEW

**Review by Stephan202 — APPROVED** [review]
Added a small commit and tweaked the suggested commit message. Thanks @mlrprananta!

File: refaster-support/src/main/java/tech/picnic/errorprone/refaster/matchers/IsRefasterAsVarargs.java [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.matchers.Matcher;
+import com.google.errorprone.refaster.Refaster;
+import com.sun.source.tree.ExpressionTree;
+
+/** A matcher of {@link Refaster#asVarargs} method invocations. */
+public final class IsRefasterAsVarargs implements Matcher<ExpressionTree> {
+  private static final long serialVersionUID = 1L;
+  private static final Matcher<ExpressionTree> DELEGATE =
+      staticMethod().onClass(Refaster.class.getName()).namedAnyOf("asVarargs");
```
Comment (rickie) [inline refaster-support/src/main/java/tech/picnic/errorprone/refaster/matchers/IsRefasterAsVarargs.java]: When passing only one, we can use `#named` 😄.
```suggestion
      staticMethod().onClass(Refaster.class.getName()).named("asVarargs");
```

File: refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsRefasterAsVarargsTest.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+            "  String[] negative3() {",
+            "    return asVarArgs(\"bar\");",
+            "  }",
+            "",
+            "  String[] positive1() {",
+            "    // BUG: Diagnostic contains:",
+            "    return Refaster.asVarargs(\"o\");",
+            "  }",
+            "",
+            " private static String[] asVarArgs(String s) {",
```
Comment (rickie) [inline refaster-support/src/test/java/tech/picnic/errorprone/refaster/matchers/IsRefasterAsVarargsTest.java:36]: Added one test case as we sometimes add this variant as well :).

**Review by rickie — APPROVED** [review]
Nice that you picked this up @mlrprananta 🚀 !

**Comment by Stephan202** [comment]
Added a commit. @rickie feel free to merge once built.

**Pull request merged**